packages/mdc-shape/README.md

```diff
@@ -0,0 +1,100 @@
+# Shape
+
+MDC Shape is a Sass / CSS / JavaScript library which draws shapes.
+
+## Design & API Documentation
+
+TODO
+
+## Installation
+
+TODO
+
+## Usage
+
+### DOM Structure
+
+```html
+<div class="mdc-shape">
+  <canvas class="mdc-shape__canvas"></canvas>
```

Comment:
Why is this a canvas and does it always have to be? Is shape expected to eventually work on other components?

Reply:
It does always have to be a canvas.

Initially we had an implementation that clipped divs with clip-path, then applied a drop-shadow filter. This has a couple performance problems
* Calculating drop-shadow requires iterating over every pixel within that div. So the larger the div the slower the performance
* The drop-shadow is repainted (and therefore recalculated) whenever the content underneath the clipped div changed. (Technically, whenever any content in the same GPU tile changes.)  This means if you were scrolling content under the clipped div, the scroll could become janky.

Also for working with other components...yes some day. Since this is still experimental we've only built some offhand prototypes (like for button). In the future most components will probably have some "shape" version, in addition to the default version, which has a modified HTML structure to include canvas. Obviously, these will be *very* heavy weight components and should be used with discretion.